### PR TITLE
CI: remove `parse_pyright_version.py` and change to simple sed script

### PR DESCRIPTION
**Pull request by AntonLydike**
This should be less brittle than before (and removes a python script from the project)

### CODE REVIEW

**Review by webmiche — APPROVED** [review]

**Review by superlopuh — APPROVED** [review]

File: .github/workflows/ci-pyright-fails.yml [hunk 1]
```diff
@@ -33,9 +33,8 @@ jobs:
       - name: Install the package locally
         run: pip install -e ".[extras]"
       - name: retrieve pyright version
-        run: |
-          python3 .github/parse_pyright_version.py
-          VER=$(cat .github/pyright-version.txt)
+        run: | # get pyright version from requirements optional file:
+          VER=$(sed -n 's/^.*pyright==\s*\(\S*\).*$/\1/p' requirements-optional.txt)
```
[the remark this answers is not in the document]
Comment (superlopuh) [reply .github/workflows/ci-pyright-fails.yml:37]: Full disclosure I asked the AI genie to write this line :)

**Review by math-fehr — APPROVED** [review]
Nice!

File: .github/workflows/ci-pyright-fails.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (AntonLydike) [reply .github/workflows/ci-pyright-fails.yml:37]: cat into grep? Whyyyyy?

[the remark this answers is not in the document]
Comment (superlopuh) [reply .github/workflows/ci-pyright-fails.yml:37]: Because that's what the genie spit out and it worked on first try :)

**Comment by AntonLydike** [comment]
@superlopuh seems like your snippet didn't work after all^^

**Comment by superlopuh** [comment]
Sad times, thank you for fixing it!

**Comment by superlopuh** [comment]
For the record it does work in a different repo so I'm not sure what the problem was :)

**Pull request merged**